### PR TITLE
Release/0.1.0

### PR DESCRIPTION
**Pull request by BenFradet**
- [x] remove `SchemaCriterionExt`
- [ ] scaladoc

### CODE REVIEW

File: src/main/scala/com.snowplowanalytics.snowplow.badrows/Failure.scala [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+package com.snowplowanalytics.snowplow.badrows
+
+import java.time.Instant
+
+import cats.data.NonEmptyList
+import cats.syntax.functor._
+import com.snowplowanalytics.iglu.client.ClientError
+import com.snowplowanalytics.iglu.core.{SchemaCriterion, SchemaKey}
+import io.circe.{Decoder, Encoder}
+import io.circe.generic.auto._
```
Comment (chuwy) [inline src/main/scala/com.snowplowanalytics.snowplow.badrows/Failure.scala]: Can we avoid `generic` derivation? I personally ran into quite a few implicit issues with it.

File: src/main/scala/com.snowplowanalytics.snowplow.badrows/Failure.scala [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+import io.circe.java8.time._
+import io.circe.syntax._
+
+sealed trait Failure
+object Failure {
+  implicit val encodeFailure: Encoder[Failure] = Encoder.instance {
```
Comment (chuwy) [inline src/main/scala/com.snowplowanalytics.snowplow.badrows/Failure.scala]: Isn't it equal to `deriveEncoder[Failure]`?

File: src/main/scala/com.snowplowanalytics.snowplow.badrows/Failure.scala [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+    Decoder[AdapterFailures].widen,
+    Decoder[SchemaViolations].widen,
+    Decoder[EnrichmentFailures].widen,
+    Decoder[SizeViolation].widen
+  ).reduceLeft(_ or _)
+}
+
+// COLLECTOR PAYLOAD FORMAT VIOLATION
+
+final case class CPFormatViolation(
```
Comment (chuwy) [inline src/main/scala/com.snowplowanalytics.snowplow.badrows/Failure.scala]: I think these should belong to `Failure` object.

File: src/main/scala/com.snowplowanalytics.snowplow.badrows/Payload.scala [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+  hostname: Option[String],
+  timestamp: Option[String], // Must have a timestamp
+  ipAddress: Option[String],
+  useragent: Option[String],
+  refererUri: Option[String],
+  headers: List[String], // Could be empty
+  networkUserId: Option[String] // User ID generated by collector-set third-party cookie
+) extends Payload
+
+final case class PartiallyEnrichedEvent(
```
Comment (chuwy) [inline src/main/scala/com.snowplowanalytics.snowplow.badrows/Payload.scala]: I thought at this time we already can assume `UUID`/`Instant`?

File: src/main/scala/com.snowplowanalytics.snowplow.badrows/BadRow.scala [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+import io.circe.{Decoder, Encoder}
+
+final case class BadRow(
+  failure: Failure,
+  payload: Payload,
+  processor: Processor
+)
+object BadRow {
+  implicit val badRowEncoder: Encoder[BadRow] = Encoder
```
[the remark this answers is not in the document]
Comment (BenFradet) [reply src/main/scala/com.snowplowanalytics.snowplow.badrows/BadRow.scala]: at compile time I would think so yes but I think I just got caught up writing encoders and decoders happy to go back to semi :)

File: src/main/scala/com.snowplowanalytics.snowplow.badrows/Failure.scala (hunk 2 above)
Comment (BenFradet) [reply src/main/scala/com.snowplowanalytics.snowplow.badrows/Failure.scala]: I don't think semiauto works with ADT but I might be wrong

File: src/main/scala/com.snowplowanalytics.snowplow.badrows/Payload.scala (hunk 4 above)
Comment (BenFradet) [reply src/main/scala/com.snowplowanalytics.snowplow.badrows/Payload.scala]: no it's already strings

File: src/main/scala/com.snowplowanalytics.snowplow.badrows/Failure.scala (hunk 3 above)
Comment (BenFradet) [reply src/main/scala/com.snowplowanalytics.snowplow.badrows/Failure.scala]: you mean:

```
sealed trait ADT
object ADT {
  final case object A extends ADT
}
```

Comment (chuwy) [reply src/main/scala/com.snowplowanalytics.snowplow.badrows/Failure.scala]: Yep

**Comment by BenFradet** [comment]
1. What I expose is `SelfDescribingData[BadRow]`
4. I think it'd be cool if it lived there

**Comment by benjben** [comment]
Awesome @chuwy , it starts to look good!!

File: src/main/scala/com.snowplowanalytics.snowplow.badrows/BadRow.scala [hunk 6]
```diff
@@ -0,0 +1,235 @@
...
+import io.circe.Json
+
+import cats.syntax.functor._
+
+import com.snowplowanalytics.iglu.core.{SelfDescribingData, SchemaKey}
+import com.snowplowanalytics.iglu.core.circe.instances._
+
+sealed trait BadRow {
+  def schemaKey: SchemaKey
+  def selfDescribinData: SelfDescribingData[Json] = SelfDescribingData(schemaKey, this.asJson)
```
Comment (dilyand) [inline src/main/scala/com.snowplowanalytics.snowplow.badrows/BadRow.scala]:
```suggestion
  def selfDescribingData: SelfDescribingData[Json] = SelfDescribingData(schemaKey, this.asJson)
```

File: src/main/scala/com.snowplowanalytics.snowplow.badrows/BadRow.scala [hunk 7]
```diff
@@ -0,0 +1,235 @@
...
+  def compact: String = selfDescribinData.asJson.noSpaces
+}
+
+object BadRow {
+  implicit val badRowEncoder: Encoder[BadRow] = Encoder.instance {
+    // Stream Collector
```
Comment (dilyand) [inline src/main/scala/com.snowplowanalytics.snowplow.badrows/BadRow.scala]: What about: https://github.com/snowplow/snowplow/blob/master/2-collectors/scala-stream-collector/core/src/main/scala/com.snowplowanalytics.snowplow.collectors.scalastream/CollectorService.scala#L283 ?

Comment (chuwy) [reply src/main/scala/com.snowplowanalytics.snowplow.badrows/BadRow.scala]: It seems I messed up the rebasing: https://github.com/snowplow/snowplow/pull/4011/files#diff-4d9d5134bd9a8a3df816fec651847ad4R277

Tests for SSC still don't compile due rebasing, but I hope I would catch it. But great point - we proably need to add another bad row type?

Comment (chuwy) [reply src/main/scala/com.snowplowanalytics.snowplow.badrows/BadRow.scala]: Apparently, it wasn't my fault - we just decided to drop it https://github.com/snowplow-incubator/snowplow-badrows/issues/24

**Pull request closed without merging**